rfcs/rfc-using-global-store-for-form-components.md

```diff
@@ -0,0 +1,317 @@
+- Start Date: 2023-03-06
+- Target Major Version: n/a
+- Reference Issues: bcgov/entity#15509
+- Entity Issue: 
+- Implementation PR: 
+
+# Summary
+
+This RFC seeks feedback to help improve the application architecture of
+business-filings-ui, business-create-ui and business-filings-ui.  At present,
+we use props and events to communicate between form components like this:
+
+```javascript
+<CourtOrderPoa
+  :draftCourtOrderNumber="getCourtOrderStep.courtOrder.fileNumber"
+  :hasDraftPlanOfArrangement="getCourtOrderStep.courtOrder.hasPlanOfArrangement"
+  :courtOrderNumberRequired="false"
+  :invalidSection="isCourtOrderInvalid"
+  @emitCourtNumber="setCourtOrderFileNumber(\$event)"
+  @emitPoa="setHasPlanOfArrangement(\$event)"
+  @emitValid="setCourtOrderValidity(\$event)"
+/>
+```
+
+We're proposing to modify the form components to use Vuex or Pinia getters and mutators instead.
```

Comment:
I agree, placing everything in a single store would make it large and unwieldy. I had envisioned separating the store into modules.  There's only three getters and mutators that are required to make most fields work:

- setNestedAttribute
- getNestedAttribute
- isNestedAttributeExist

The SingleCheckbox.vue is a special case because checking a box, adds or removes items from an array -- it doesn't mutate a single attribute.  Now that you mention it, those two actions, `addItemToCheckBoxList` and `deleteItemFromCheckboxList` should be moved back to the SingleCheckbox component as I don't think they'd be used by other fields.

Reply:
Yes, you are correct, the store structure is predefined ahead of time.  I like your idea of having setNestedAttribute validate the type as a safety check.